scene-lib/src/scenelib/annotations/AnnotationBuilder.java

```diff
@@ -209,28 +212,34 @@ public Annotation finish() {
         active = false;
         if (def == null) {
             assert fieldTypes != null;
-            def = new AnnotationDef(typeName, tlAnnotationsHere, fieldTypes);
+            def = new AnnotationDef(typeName, tlAnnotationsHere, fieldTypes, source);
         } else {
             assert typeName == null;
             assert fieldTypes.isEmpty();
         }
         return new Annotation(def, fieldValues);
     }
 
-    AnnotationBuilder(AnnotationDef def) {
+    AnnotationBuilder(AnnotationDef def, String source) {
         assert def != null;
+        assert source != null;
         this.def = def;
+        this.source = source;
     }
 
     AnnotationBuilder(String typeName) {
```

Comment:
Should this constructor also get a `String source` parameter? Otherwise, field `source` will always be null and the assert will fail.